File: src/ru/gurps/generator/models/rules/Addon.java
package ru.gurps.generator.models.rules;

import ru.gurps.generator.config.Model;

public class Addon extends Model {
    public Integer id;
    public Integer featureId;
    public String name;
    public String nameEn;
    public String cost;
    @Ignore public Integer resultCost = 0;
    public String description;
    public Integer maxLevel;
    @Ignore public String level = "1";
    @Ignore public Boolean active = false;

    public Addon() {
    }

    public Addon(Integer id, Integer featureId, String name, String nameEn, String cost, Integer resultCost, String description, Integer maxLevel, Boolean active, String level) {
        this.id = id;
        this.featureId = featureId;
        this.name = name;
        this.nameEn = nameEn;
        this.cost = cost;
        this.resultCost = resultCost;
        this.description = description;
        this.maxLevel = maxLevel;
        this.active = active;
        this.level = level;
    }

    public String getName() {
        return name;
    }

    public String getNameEn() {
        return nameEn;
    }

    public String getCost() {
        return cost;
    }

    public String getLevel() {
        return level;
    }
}

File: src/main/kotlin/ru/gurps/generator/desktop/models/rules/Addon.kt
package ru.gurps.generator.desktop.models.rules

import ru.gurps.generator.desktop.interfaces.models.Ability
import ru.gurps.generator.desktop.config.Model
import java.sql.ResultSet

class Addon : Model, Ability {
    override var id: Int = -1
    var featureId: Int = -1
    override var name: String = ""
    override var nameEn: String = ""
    override var cost: Int = 0
    override var currentCost: Int = 0
    override var description: String = ""
    var maxLevel: Int = 0
    var level = 1
    override var add: Boolean = false

    constructor()
    constructor(result: ResultSet) {
        id = result.getInt("id")
        featureId = result.getInt("featureId")
        val nameRes = result.getString("name")
        if (nameRes != null) name = nameRes
        val nameEnRes = result.getString("nameEn")
        if (nameEnRes != null) nameEn = nameEnRes
        cost = result.getInt("cost")
        currentCost = result.getInt("cost")
        val descriptionRes = result.getString("description")
        if (descriptionRes != null) description = descriptionRes
        maxLevel = result.getInt("maxLevel")
    }

    fun getLevel(): String {
        return level.toString()
    }

    fun getCost(): String {
        return "$currentCost%"
    }
}
